modules/sdk/src/main/scala/org/tessellation/sdk/domain/snapshot/services/GlobalL0Service.scala

```diff
@@ -37,20 +50,36 @@ object GlobalL0Service {
     l0GlobalSnapshotClient: L0GlobalSnapshotClient[F],
     globalL0ClusterStorage: L0ClusterStorage[F],
     lastGlobalSnapshotStorage: LastSnapshotStorage[F, GlobalIncrementalSnapshot, GlobalSnapshotInfo],
-    singlePullLimit: Option[PosLong]
+    singlePullLimit: Option[PosLong],
+    maybeMajorityPeerIdSet: Option[NonEmptySet[PeerId]]
   ): GlobalL0Service[F] =
     new GlobalL0Service[F] {
 
       private val logger = Slf4jLogger.getLogger[F]
+      private val maybeMajorityPeerIds = maybeMajorityPeerIdSet.map(_.toNonEmptyList)
 
       def pullLatestSnapshot: F[(Hashed[GlobalIncrementalSnapshot], GlobalSnapshotInfo)] =
```

Comment:
Maybe we should now explicitly mark these functions as downloading randomly. So `randomlyPullLatestSnapshot` or similar. We could have the functions that switch between random and majority based on config be called `pullLatestSnapshot` and use either `randomlyPull` or `pullMajority` the way I suggested in another comment.

Reply:
@TheMMaciek Correct me if I'm wrong but aren't both versions randomly pulling from a peer? The difference with the _majority_ functions is that the selected peer's snapshot hash has to match the one determined by the majority function; when it doesn't we choose another peer.

Reply:
Yeah in a way both use some random factor. We can figure out a better name, whatever that name might be. What I wanted to achieve is to simply give some idea on what is the difference between these two strategies of pulling snapshot and make that more apparent in the name of the function.

Additionally we could also think about whether we want to expose them to the user so have `pullMajority...` and `randomlyPull...` available publicly, or we would rather only expose `pullSnapshots...` that switch between those two strategies based on whether the majority list is present or not. If we want to expose them then we may need to make the majority function throw an error if a user tries to use it but a majority list is not supplied.

Reply:
Renaming these seems fine to me, what we're doing here is a random pull against a known majority. It can be renamed to indicate that, so in the future if we add different functions (i.e. pull random with no check, or pull from majority with no random load,) then that would indicate the difference. I don't think we need to cover every case here (i.e. writing all the variations,) so long as we're naming it to indicate there are other variations possible.